apps/etterlatte-brev-api/src/main/kotlin/no/nav/etterlatte/brev/vedtaksbrev/VedtaksbrevServiceNy.kt

```diff
@@ -0,0 +1,286 @@
+        bruker: BrukerTokenInfo,
+        brevRequest: BrevRequest,
+    ): Pdf {
+        val brev = db.hentBrev(brevId)
+
+        val brevInnholdData = utledBrevInnholdData(brev, brevRequest)
+
+        // TODO Må dette hentes på nytt ved forhåndsvisning? Det kan endre seg?
+        val avsender =
+            utledAvsender(bruker, brevRequest.saksbehandlerIdent, brevRequest.attestantIdent, brevRequest.sak.enhet)
+
+        val pdf = opprettPdf(brev, brevRequest, brevInnholdData, avsender)
+
+        // TODO ferdigstilles vel aldri her?
+        logger.info("PDF generert ok. Sjekker om den skal lagres og ferdigstilles")
+        brev.brevkoder?.let { db.oppdaterBrevkoder(brev.id, it) }
+        if (brevRequest.skalLagre) {
+            logger.info("Lagrer PDF for brev med id=$brevId")
+            db.lagrePdf(brevId, pdf)
+        }
+
+        return pdf
+    }
+
+    fun ferdigstillVedtaksbrev(
+        behandlingId: UUID,
+        brukerTokenInfo: BrukerTokenInfo,
+    ) {
+        val brev =
+            krevIkkeNull(hentVedtaksbrev(behandlingId)) {
+                "Fant ingen brev for behandling (id=$behandlingId)"
+            }
+
+        if (brev.status == Status.FERDIGSTILT) {
+            logger.warn("Brev (id=${brev.id}) er allerede ferdigstilt. Avbryter ferdigstilling...")
+            return
+        } else if (!brev.kanEndres()) {
+            throw UgyldigStatusKanIkkeFerdigstilles(brev.id, brev.status)
+        } else if (brev.mottakere.size !in 1..2) {
+            logger.error("Brev ${brev.id} har ${brev.mottakere.size} mottakere. Dette skal ikke være mulig...")
+            throw UgyldigAntallMottakere()
+        } else if (brev.mottakere.any { it.erGyldig().isNotEmpty() }) {
+            sikkerlogger.error("Ugyldig mottaker(e): ${brev.mottakere.toJson()}")
+            throw UgyldigMottakerKanIkkeFerdigstilles(brev.id, brev.sakId, brev.mottakere.flatMap { it.erGyldig() })
+        }
+
+        logger.info("Ferdigstiller brev med id=${brev.id}")
+        if (db.hentPdf(brev.id) == null) {
+            throw BrevManglerPDF(brev.id)
+        } else {
+            db.settBrevFerdigstilt(brev.id, brukerTokenInfo)
+        }
+    }
+
+    suspend fun tilbakestillVedtaksbrev(
+        brevId: Long,
+        bruker: BrukerTokenInfo,
+        brevRequest: BrevRequest,
+    ): BrevService.BrevPayload {
+        val brev = db.hentBrev(brevId)
+        if (!brev.kanEndres()) {
+            throw UgyldigForespoerselException(
+                "BREVET_KAN_IKKE_ENDRES",
+                "Kan ikke oppdatere brevet med id=$brevId, fordi brevet har status (${brev.status})",
+            )
+        }
+
+        val (spraak, sak, innsender, soeker, avdoede, verge, saksbehandlerIdent, attestantIdent, _, brevInnholdData) = brevRequest
+
+        val brevKode = brevInnholdData.brevKode
+        val avsender = utledAvsender(bruker, saksbehandlerIdent, attestantIdent, sak.enhet)
+
+        // val spraak = db.hentBrevInnhold(brevId)?.spraak TODO ??
```

Comment:
Jeg tenker det å hente språk fra databasen her er bedre, hvis vi støtter den flyten. Men når jeg tenker meg om krever nok det en del endringer i flyten. Jeg lurer på om vi bør ta endring av språk i vedtaksbrev også til behandling-brev-routen, siden vi vil trenge å regenerere det redigerbare innholdet for andre brevtyper